### PR TITLE
connector: filter OIDC identities w/ configured domain

### LINKED ISSUE #66: cmd/dex: user filters
We'd like to be able to restrict login to a service to any valid google account, for example `@coreos.com`.

### PR DESCRIPTION
**Pull request by bcwaldon**
Fixes #66

### CODE REVIEW

**Comment by bcwaldon** [comment]
@bobbyrullo looking for input here on how to best test the changes to handleCallbackFunc

File: connector/connector_oidc.go [hunk 1]
```diff
@@ -37,6 +40,37 @@ func (cfg *OIDCConnectorConfig) ConnectorType() string {
...
+		if err != nil {
+			return err
+		} else if want != got {
+			return errors.New("domain mismatch")
+		}
+		return nil
+	}
+}
+
+func parseEmailDomain(email string) (string, error) {
```
Comment (bcwaldon) [inline connector/connector_oidc.go]: I don't like that I had to write this function, but this is all I could find in stdlib: https://golang.org/pkg/net/mail/#ParseAddress

File: Documentation/connectors-configuration.md [hunk 2]
```diff
@@ -52,6 +52,8 @@ This connector config lets users authenticate with other OIDC providers. In addi
 
 * trustedEmailProvider: a `boolean`. If true dex will trust the email address claims from this provider and not require that users verify their emails.
 
+* domain: a `string` (optional). If set to a non-empty string, users that are unable to log in to the remote Identity Provider with an email address whose domain matches the provided value will be prevented from logging in to or registering with dex. For example, if `domain` is set to `example.com`, user `foo@example.com` will be granted an identity token from dex, but `bar@example.net` will not.
```
Comment (joshix) [inline Documentation/connectors-configuration.md]: * domain: a `string` (optional). If set, users' email address domain must match the provided value. Email addresses that do not match will not be allowed to register or log in with dex. For example, if `domain` is set to `example.com`, the email address `user@example.com` will be granted an identity token, but `user@example.net` will not.

(an attempt to clarify a graf that confused me)

**Comment by bcwaldon** [comment]
Closing this for now. See https://github.com/coreos/dex/issues/66#issuecomment-164910467

**Pull request closed without merging**